src/pkcs11.cpp

```diff
@@ -204,6 +235,9 @@ int pkcs11_create_csr(const lmp_options &opt, string &pkey, string &csr)
 	if (PKCS11_login(slot, 0, opt.hsm_pin.c_str()))
 		leave;
 
+	if (remove_keys(slot->token, hsm_cfg.tls_lbl))
```

Comment:
If an initial registration was successful, and then for some reason a user decides to re-register a device again, and the re-registration fails then a device ends up without valid "keys".
I am not sure that this is critical, because a device ultimately gets valid "keys" once it is successfully re-registered.

Reply:
The user would have had to remove `/var/sota/` content in order to be able to try to re-register, so I believe removing the keys is fine, even if something goes wrong later on.
I should probably remove existing certificates with "client" label as well, since they would be referring to key pairs that are now removed, and won't be usable anymore. I'll add that once I'm back later today.
@ldts any comments on that and on the general fix? I can't add reviewers to the PR.

Reply:
>The user would have had to remove /var/sota/ content in order to be able to try to re-register, so I believe removing the keys is fine, even if something goes wrong later on.

Not exactly, this is enough to remove just `sql.db` to re-register a device. 
Having said that I don't see any use-case when removing the key can cause some trouble providing that we document this detail ("device re-registration is not an atomic transaction, its failure doesn't mean that a device is the same state as before the re-registration").

Maybe @doanac can think of an use-case when it can be critical.

Reply:
I think from a validation standpoint, we have to protect for this:

 * lmp-device-register runs successfully. aklite daemon is up
 * time goes by
 * aklite is stopped for some reason. At this point people tend to re-run lmp-device-register w/o really thinking through the consequences. As long as lmp-device-register exits here cleanly and doesn't remove keys we are fine.
 * user does an `rm /var/sota/sql.db` cause they really want to re-register
 * lmp-device-register should be able to work and generate new keys

Reply:
>aklite is stopped for some reason. At this point people tend to re-run lmp-device-register w/o really thinking through the consequences. As long as lmp-device-register exits here cleanly and doesn't remove keys we are fine.

It is so, as long as `sql.db` is not removed.

Reply:
>user does an rm /var/sota/sql.db cause they really want to re-register
> lmp-device-register should be able to work and generate new keys

What if it doesn't work (re-registration fail), is it fine that the key is removed?
I think, it is OK. Since a user removes `sql.db` he/she is in "advanced" mode and should continue with device re-registration attempts until it succeeds.

Reply:
Yeah, at this stage the user is already doing a more advanced flow. Without sql.db we can't really know what to do.

I know there could be use cases such as when the user performs a factory reset but decides to keep the hsm specific keys, but for that we would need to restore sota.toml and currently sql.db as well in order for the device to come up in a managed stage again after factory reset.

Saying that, I think the current logic here is fine.